specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/StackHCI/edgeMachineJob.tsp

```diff
@@ -0,0 +1,761 @@
+
+  @doc("Location of the secret used for authentication.")
+  sshPubKey?: string[];
+}
+
+@added(Versions.v2025_12_01_preview)
+@doc("Onboarding resource type.")
+union OnboardingResourceType {
+  string,
+
+  @doc("Hybrid Compute Machine.")
+  HybridComputeMachine: "HybridComputeMachine",
+}
+
+@added(Versions.v2025_12_01_preview)
+@doc("Onboarding configuration.")
+model OnboardingConfiguration {
+  @doc("Type of the onboarding resource to support polymorphic resource.")
+  @visibility(Lifecycle.Create, Lifecycle.Read)
+  type?: OnboardingResourceType;
+
+  @doc("Resource ID.")
+  resourceId?: Azure.Core.armResourceIdentifier;
+
+  @doc("Location of the resource.")
+  @visibility(Lifecycle.Create, Lifecycle.Read)
+  location?: string;
+
+  @doc("Tenant ID of the resource.")
+  tenantId?: string;
+
+  @doc("Azure Arc virtual machine ID.")
+  arcVirtualMachineId?: string;
+}
+
+@added(Versions.v2025_12_01_preview)
+@doc("Device configuration.")
+model DeviceConfiguration {
+  @doc("Network configuration.")
+  network?: NetworkConfiguration;
+
+  @doc("Hostname of the device.")
+  hostName?: string;
+
+  @doc("Web proxy configuration.")
+  webProxy?: WebProxyConfiguration;
+
+  @doc("Time configuration.")
+  time?: TimeConfiguration;
+
+  @doc("Storage configuration.")
+  storage?: StorageConfiguration;
+}
+
+@added(Versions.v2025_12_01_preview)
+@doc("Network configuration.")
+model NetworkConfiguration {
+  @Azure.ResourceManager.identifiers(#[])
+  @doc("List of network adapters.")
+  networkAdapters?: NetworkAdapter[];
+}
+
+@added(Versions.v2025_12_01_preview)
+@doc("Storage configuration.")
+model StorageConfiguration {
+  @doc("Partition size.")
+  partitionSize?: string;
+}
+
+@added(Versions.v2025_12_01_preview)
+@doc("Network adapter configuration.")
+model NetworkAdapter {
+  @doc("Type of IP assignment.")
+  ipAssignmentType: IpAssignmentType;
+
+  @doc("IP address.")
+  ipAddress?: string;
+
+  @doc("Adapter Name.")
+  adapterName?: string;
+
+  @doc("MAC address.")
+  macAddress?: string;
+
+  @doc("IP address range.")
+  ipAddressRange?: IpAddressRange;
+
+  @doc("Gateway id.")
+  gateway?: string;
+
+  @doc("Subnet mask.")
+  subnetMask?: string;
+
+  @doc("Array of DNS addresses.")
+  dnsAddressArray?: string[];
+
+  @doc("VLAN ID for the network setup.")
+  vlanId?: string;
+}
+
+@added(Versions.v2025_12_01_preview)
+@doc("IP assignment types")
+union IpAssignmentType {
+  string,
+
+  @doc("Automatic IP assignment")
+  Automatic: "Automatic",
+
+  @doc("Manual IP assignment")
+  Manual: "Manual",
+}
+
+@added(Versions.v2025_12_01_preview)
+@doc("IP address range configuration.")
+model IpAddressRange {
+  @doc("Start IP address.")
+  startIp: string;
+
+  @doc("End IP address.")
+  endIp: string;
+}
+
+@added(Versions.v2025_12_01_preview)
+@doc("Web proxy configuration.")
+model WebProxyConfiguration {
+  @doc("Connection URI of the web proxy.")
+  connectionUri?: string;
+
+  @doc("Port of the web proxy.")
+  port?: string;
```

Comment:
>port?: string; [](http://example.com/codeflow?start=2&length=14)

isn't there any pattern needed for this?

Reply:
added pattern